### PR TITLE
Review pull request: BXMSDOC-5099 New CaseLogCleanupCommand command

### PR DESCRIPTION
**Pull request by michelehaglund**
Original jira: https://issues.redhat.com/browse/BXMSDOC-5099
Rendered output: http://file.rdu.redhat.com/~mhaglund/BXMSDOC-5099/#case-management-clean-case-log-proc

### CODE REVIEW

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,61 @@
+[id='case-management-clean-case-log-proc']
+= Removing case logs from the database
+
+Use the `CaseLogCleanupCommand` to cleanup cases, such as canceled cases that lack value and that are using database space. The `CaseLogCleanupCommand` command consists of logic to clean up all or selected cases automatically. By default, the `CaseLogCleanupCommand` is executed once a day but can be configured to run on different intervals.
```
Comment (MarianMacik) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: > By default, the `CaseLogCleanupCommand` is executed once a day but can be configured to run on different intervals.

Where do we have this information from? I am not aware that it is configured this way by default.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+| Name
+| Description
+| Is Exclusive
+
+| SkipProcessLog
+| Indicates if the clean-up of process instances, node instances and variables log cleanup should be omitted (default: false)
+| No, can be used with other parameters
+
+| SkipTaskLog
+| Indicates if the task audit and the task event log clean-up should be omitted (default: false)
```
[the remark this answers is not in the document]
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Updated.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+| OlderThan
+| Causes logs older than the given date to be removed. The date format is `YYYY-MM-DD`. Usually, this parameter is used for single run jobs
+| Yes, cannot be used when the OlderThanPeriod parameter is used
+
+| OlderThanPeriod
+| Causes logs older than the given timer expression should be removed. For example, set 30d to remove logs older than 30 day from current time
+| Yes, cannot be used when the OlderThan parameter is used
+
+| ForCaseDefId or similar
```
[the remark this answers is not in the document]
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Removed "or similar"

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+| Specifies the case definition ID for which the logs should be removed
+| No, can be used with other parameters
+
+| ForDeployment
+| Specifies deployment ID for which logs should be removed
+| No, can be used with other parameters
+
+| EmfName
+| Persistence unit name that shall be used to perform operation deletion
```
[the remark this answers is not in the document]
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Add default

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+| N/A
+
+| Status
+| Status of the case instances for which logs should be removed
+| No, can be used with other parameters
+|===
```
[the remark this answers is not in the document]
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Added.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 1 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Carried over from:

https://access.redhat.com/documentation/en-us/red_hat_jboss_bpm_suite/6.2/html/administration_and_configuration_guide/configuring_logging3

Removed default sentence.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 6]
```diff
@@ -13,49 +13,53 @@ There are several important configuration options that can be used with the `Cas
 | Is Exclusive
 
 | SkipProcessLog
-| Indicates if the clean-up of process instances, node instances and variables log cleanup should be omitted (default: false)
+| Indicates if the of process and node instances, and variable logs clean-up should be omitted. (default: false)
```
Comment (MarianMacik) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Probably typo in `Indicates if the of process and node instances`.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 1 above)
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: What does "lack value" mean? Literally has no values in it or that it is not relevant anymore?
"using up database space"?
The command consists of logic or contains logic?

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 7]
```diff
@@ -0,0 +1,61 @@
+[id='case-management-clean-case-log-proc']
+= Removing case logs from the database
+
+Use the `CaseLogCleanupCommand` to cleanup cases, such as canceled cases that lack value and that are using database space. The `CaseLogCleanupCommand` command consists of logic to clean up all or selected cases automatically. By default, the `CaseLogCleanupCommand` is executed once a day but can be configured to run on different intervals.
+
+There are several important configuration options that can be used with the `CaseLogCleanupCommand` command:
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: "that you can use with"?

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+.CaseLogCleanupCommand parameters table
+[cols="1,1,1", options="header"]
+|===
+| Name
+| Description
+| Is Exclusive
+
+| SkipProcessLog
+| Indicates if the clean-up of process instances, node instances and variables log cleanup should be omitted (default: false)
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: "should be omitted" or "is omitted"?  I am guessing that if this parameter is flagged then the process & node instances along with the variable log clean up will be skipped when the command runs.

Same comment throughout.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+| No, can be used with other parameters
+
+| SingleRun
+| Indicates if the job routine should run only once (default: false)
+| No, can be used with other parameters
+
+| NextRun
+| Sets a date for the next run.
+
+For example, 12h is set for jobs to be executed every 12 hours.
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Who sets this? If the audience for this topic needs to do so, make this imperative.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+If the option is not given, the next job will run 24 hours after the completion of the current job
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: If the value is not set?

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 6 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Fixed. Thank you!

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+| No, can be used with other parameters
+
+| OlderThan
+| Causes logs older than the given date to be removed. The date format is `YYYY-MM-DD`. Usually, this parameter is used for single run jobs
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: given date or specified date?

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+| Yes, cannot be used when the OlderThanPeriod parameter is used
+
+| OlderThanPeriod
+| Causes logs older than the given timer expression should be removed. For example, set 30d to remove logs older than 30 day from current time
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Given or specified?

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+| Yes, cannot be used when the OlderThan parameter is used
+
+| ForCaseDefId or similar
+| Specifies the case definition ID for which the logs should be removed
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: I am having difficulty with this sentence. "of the logs that should be removed"?

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+| No, can be used with other parameters
+
+| ForDeployment
+| Specifies deployment ID for which logs should be removed
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Same thing. It reads weird.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 4 above)
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: The "shall" proclamation is eng speak. Can we use "is" instead?

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+| No, can be used with other parameters
+
+| EmfName
+| Persistence unit name that shall be used to perform operation deletion
+| N/A
+
+| Status
+| Status of the case instances for which logs should be removed
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Fix this also.

File: doc-content/enterprise-only/cases/case-management-closing-cases-ref.adoc [hunk 16]
```diff
@@ -2,6 +2,8 @@
 = Closing cases
 
 A case instance can be completed when there are no more activities to be performed and the business goal is achieved, or it can be closed prematurely. Usually the case owner closes the case when all work is completed and the case goals have been met.
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-closing-cases-ref.adoc]: Why are we telling people that you can close it prematurely? Is there a way to tell? What should they do if it is closed before it ought to be?

File: doc-content/enterprise-only/cases/case-management-closing-cases-ref.adoc [hunk 17]
```diff
@@ -2,6 +2,8 @@
...
-When you close a case, consider adding a comment about why the case instance is being closed. A closed case can be reopened later with the same case ID if required.
+When you close a case, consider adding a comment about why the case instance is being closed.
```
Comment (pnagesha) [inline doc-content/enterprise-only/cases/case-management-closing-cases-ref.adoc]: why you are closing?

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 7 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Changed as suggested.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 1 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Rewritten as: 

Use the `CaseLogCleanupCommand` to clean-up cases, such as canceled cases that are using up database space. The `CaseLogCleanupCommand` command contains logic to automatically clean-up all or selected cases.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 8 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Reworded:
Indicates whether or not the process and node instances, along with the process variable log clean-up will be skipped when the command runs. Default value: `false`

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 10 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Emailed dev for default value - if there is one.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 9 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Reworded.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 10 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Note that, If the option is not given, the next job will run 24 hours after the completion of the current job
Marian to clarify if 24 hrs is the default

Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: For now:

NextRun
| Sets a date for the next run. For example, 12h is set for jobs to be executed every 12 hours. The ISO format can be used to set the precise date. If no value is provided, the next job will run 24 hours after the completion of the current job.
| No, can be used with other parameters

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 11 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Went with "specified" :)

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 12 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Specified:

OlderThanPeriod
| Logs older than the specified timer expression are removed. For example, set 30d to remove logs older than 30 days.
| Yes, cannot be used when the `OlderThan` parameter is used

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 13 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Simplified:

Specifies the case definition ID of the logs that are removed.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 14 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Updated

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 4 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: Already changed

File: doc-content/enterprise-only/cases/case-management-closing-cases-ref.adoc (hunk 16 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-closing-cases-ref.adoc]: Good question. Sent the same out for QE to clarify. I can ASSuME why, but I will check 1st.

File: doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc (hunk 10 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/cases/case-management-clean-case-log-proc.adoc]: | NextRun
| Schedules the next job execution. For example, set to `12h` for jobs to be executed every 12 hours. The schedule is ignored if you set *SingleRun* to `true`, unless you set both *SingleRun* and *NextRun*. If both are set, the *NextRun* schedule takes priority. The ISO format can be used to set the precise date. Default value: `24h`
| No, can be used with other parameters

**Pull request merged**